packages/core/src/context/index.ts

```diff
@@ -0,0 +1,125 @@
+import { CoreSegmentEvent } from '../events/interfaces'
+
+import { v4 as uuid } from '@lukeed/uuid'
+import { dset } from 'dset'
+import { CoreLogger, LogLevel, LogMessage } from '../logger'
+import Stats, { Metric } from '../stats'
+
+export function isContext(v?: object): v is CoreContext {
```

Comment:
This function is new to allow duck typing as well as instanceOf -- I thought it would be good to add the edge case, but it's not critical